src/tools/registry/gen_inc.c

```diff
@@ -83,6 +109,122 @@ int write_field_pointer_arrays(FILE* fd){/*{{{*/
 	return 0;
 }/*}}}*/
 
+// Checks for a string in a list of strings.
+// Returns the index of the string if it does exist in the array,
+// and -1 if it does not appear in the array.
+int find_string_in_array(char *input_string, const char *array[], size_t rows){
+	size_t i;
+	for (i = 0; i < rows; i++ ){
+		if (strcmp(input_string, array[i]) == 0){
+			return i;
+		}
+	}
+	return -1;
+}
+
+// Helper function to change attribute names in accordance with
+// "attrs_to_modify" within the "add_attribute_if_not_ignored" function
+const char * modify_attr(const char *attr, const char *array[][2], size_t rows) {
+	size_t i;
+	for (i = 0; i < rows; i++) {
+		if (strcmp(attr, array[i][0]) == 0) {
+			return array[i][1];
+		}
+	}
+	return attr;
+}
+
+
+// Doubles single quotes in stringIn, and places the results in the buffer
+// stringOut. stringOut should be large enough to store (len(stringIn) * 2 + 1)
+// characters. Returns 1 if the buffer is too small for the result. 
+int escape_quotes(const char * stringIn, char * result, size_t bufferSize){
+	size_t resultIndex = 0;
+	
+	for (size_t i = 0; i < strlen(stringIn); i++) {
+		if ( stringIn[i] == '\'' ) {
+			if (resultIndex + 1 > bufferSize) return 1;
```

Comment:
This is probably a matter of personal preference, but if you change the condition to `resultIndex >= bufferSize`, then there's no need for arithmetic.